### PR TITLE
Fix some bugs with the docker scripts.

### PR DESCRIPTION
**Pull request by jeffmay**
@audax-shreyaspurohit @mingy711 @macmacbr @danstaley @egprentice

Fixed some issues with the docker scripts.

### CODE REVIEW

File: docker/install.sh [hunk 1]
```diff
@@ -43,23 +32,24 @@ key="$1"
 case $key in
 
     -d|--driver)
-        DOCKER_DRIVER="$2"
+        if [[ "$2" != "" ]]; then
+            DOCKER_DRIVER="$2"
+        fi
```
Comment (audax-shreyaspurohit) [inline docker/install.sh:37]: this is fine, just fyi, getopts can also be used- http://wiki.bash-hackers.org/howto/getopts_tutorial

**Comment by audax-shreyaspurohit** [comment]
lgtm

File: docker/install.sh [hunk 2]
```diff
@@ -73,19 +63,37 @@ esac
...
 
-# Get the latest docker image for neo4j
-docker pull neo4j:${NEO4J_VERSION}
+# Download the docker image for Neo4j
+if [[ "$(docker images | grep neo4j | grep ${NEO4J_VERSION})" ]]; then
+    echo "docker-machine env '$DOCKER_ENV' already has the neo4j $NEO4J_VERSION image available."
+else
+    docker pull neo4j:${NEO4J_VERSION}
+fi
+echo "Run bin/start.sh to start Neo4j with this container."
```
Comment (jeffmay) [inline docker/install.sh]: FAIL: Should be "docker/start.sh" and it is an image not a container.

**Comment by macmacbr** [comment]
LGTM.   gotta love the "Coverage decreased (-0.0%) to 68.369%"

File: docker/install.sh [hunk 3]
```diff
@@ -73,19 +63,37 @@ esac
...
 done
 
-# Verify that the driver is provided
+# Set the defaults
 if [[ "$DOCKER_DRIVER" == "" ]]; then
-    echo "Argument -d | --driver is required. Please see options from docker-machine:"
-    echo ""
-    echo $(docker-machine create --help | grep "\-\-driver")
-    exit 1
+    DOCKER_DRIVER=${DEFAULT_DOCKER_DRIVER}
```
Comment (macmacbr) [inline docker/install.sh:68]: HINT: on bash 4 there's a format to set the default on a variable and print a message, but this is fine too.

**Pull request merged**